### DISCUSSION TITLE
Markdown is not perfectly rendered in remix vite

### DISCUSSION THREAD

**Discussion by mhanifmuhsin** [mdx-js/mdx#2522] [Q&A] [answered]
I have a problem with markdown being rendered imperfectly, for example the code doesn't automatically scroll when it is responsive, I built this website with remix, here is the website you can visit [my website](https://frontendme.web.id/blog/first)

<img width="500" alt="Screenshot 2024-08-06 at 23 15 30" src="https://github.com/user-attachments/assets/579c99d3-1758-4ede-b877-ffe8c6694684">

![code](https://github.com/user-attachments/assets/e4ea2879-1a44-41c2-8d2c-246a70e81804)

and this is the vite config that I did
```bash
import { vitePlugin as remix } from "@remix-run/dev";
import { defineConfig } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import mdx from "@mdx-js/rollup";
import remarkFrontmatter from "remark-frontmatter";
import remarkMdxFrontmatter from "remark-mdx-frontmatter";
import rehypePrettyCode from "rehype-pretty-code";

export default defineConfig({
	plugins: [
		mdx({
			remarkPlugins: [remarkFrontmatter, remarkMdxFrontmatter],
			rehypePlugins: [rehypePrettyCode],
		}),
		remix({
			future: {
				v3_fetcherPersist: true,
				v3_relativeSplatPath: true,
				v3_throwAbortReason: true,
			},
		}),
		tsconfigPaths(),
	],
});
```
I would be very grateful if there are friends there who have a solution to this problem.

**Comment by ChristianMurphy** [comment]
Welcome @mhanifmuhsin! 👋 

MDX does not provide styles, what you are seeing is a stylesheet issue.
The issue you are seeing looks like a tailwind issue.

Rehype pretty code block recommends these styles https://rehype-pretty.pages.dev/#styles
If you have more questions on how to style rehype-pretty-code, their community may be able to offer more insights https://github.com/rehype-pretty/rehype-pretty-code/discussions

**Reply by mhanifmuhsin** [reply]
ok, thank you for your response, but I have another question, this is 1 example when I made code **1. sample** with spaces, like this : 

<img width="1660" alt="Screenshot 2024-08-07 at 08 23 05" src="https://github.com/user-attachments/assets/b489e077-ef37-4089-9250-60df2f72cf8f">

if I do it in Playground MDX it successfully renders perfectly, but when I do it in my coding,

<img width="859" alt="Screenshot 2024-08-07 at 08 28 04" src="https://github.com/user-attachments/assets/69d794b5-56ca-428f-809f-393f281105e6">
<img width="1125" alt="Screenshot 2024-08-07 at 08 28 30" src="https://github.com/user-attachments/assets/0126909d-01f8-4e8e-ba9f-423590923a91">

 this one is not perfect, is it This is the same as the tailwind problem ?

**Reply by ChristianMurphy** [reply, marked as answer]
> This is the same as the tailwind problem?

Yes

**Reply by mhanifmuhsin** [reply]
ok, thank you very much @ChristianMurphy  😊

**Reply by mhanifmuhsin** [reply]
For fix problem I have found the solution of :
[Rehype Pretty](https://rehype-pretty.pages.dev/)
[Git Rehype Pretty](https://github.com/rehype-pretty/rehype-pretty-code/blob/master/examples/next/tailwind.config.js)
[Tailwinnd CSS Typography](https://github.com/tailwindlabs/tailwindcss-typography)
and then this is [my website](https://frontendme.web.id/blog/first)
